core/src/subgraph/instance_manager.rs

```diff
@@ -156,6 +156,17 @@ impl SubgraphInstanceManager {
         let block_logger = logger.clone();
         let error_logger = logger.clone();
 
+        // Clear the 'failed' state of the subgraph. We were told explicitly
+        // to start, which implies we assume the subgraph has not failed (yet)
+        let status_ops = SubgraphDeploymentEntity::update_failed_operations(&id_for_err, false);
+        if let Err(e) = store_for_errors.apply_entity_operations(status_ops, EventSource::None) {
+            error!(
+                error_logger,
+                "Failed to clear 'failed' status of subgraph: {}", e;
+                "id" => id_for_err.to_string()
+            );
+        }
```

Comment:
One thing I wonder is if we should even continue starting the subgraph if clearing the failed status fails. I'm thinking not.

Reply:
I agree; I changed things so that the error is just returned back to the caller in `SubgraphInstanceManager::handle_subgraph_events` which will log it.